cli/cmd/inject.go

```diff
@@ -56,10 +57,21 @@ with 'conduit inject'. e.g. curl http://url.to/yml | conduit inject -
 				return err
 			}
 		}
-		return InjectYAML(in, os.Stdout)
+		return runInjectCmd(in, os.Stderr, os.Stdout)
 	},
 }
 
+func runInjectCmd(input io.Reader, errWriter io.Writer, outWriter io.Writer) error {
+	postInjectBuf := &bytes.Buffer{}
+	err := InjectYAML(input, postInjectBuf)
+	if err != nil {
+		fmt.Fprintf(errWriter, "Error injecting conduit proxy: %v", err)
+		return nil //Return nil so we do not print usage information in the command
+	}
+	io.Copy(outWriter, postInjectBuf)
```

Comment:
You need to check the return value of Copy and exit with a non-zero exit code on error.